nautobot/core/celery/schedulers.py

```diff
@@ -111,8 +117,44 @@ def apply_async(self, entry, producer=None, advance=True, **kwargs):
 
         Ref: https://github.com/celery/django-celery-beat/issues/558#issuecomment-1162730008
         """
-        # TODO expand this instead of calling super()
-        resp = super().apply_async(entry, producer=producer, advance=advance, **kwargs)
+        resp = None
+        entry = self.reserve(entry) if advance else entry
+        task = self.app.tasks.get(entry.task)
+
+        try:
+            entry_args = _evaluate_entry_args(entry.args)
+            entry_kwargs = _evaluate_entry_kwargs(entry.kwargs)
+            if task:
+                scheduled_job = entry.model
+                job_queue = scheduled_job.job_queue
+                # Distinguish between Celery and Kubernetes job queues
+                if job_queue.queue_type == JobQueueTypeChoices.TYPE_KUBERNETES:
+                    job_result = JobResult.objects.create(
+                        name=scheduled_job.job_model.name,
+                        job_model=scheduled_job.job_model,
+                        scheduled_job=scheduled_job,
+                        user=scheduled_job.user,
+                    )
+                    job_result = run_kubernetes_job_and_return_job_result(
+                        job_queue, job_result, json.dumps(entry_kwargs)
+                    )
+                    # Return an AsyncResult object to mimic the behavior of Celery tasks after the job is finished by Kubernetes Job Pod.
+                    resp = AsyncResult(job_result.id)
+                else:
+                    resp = task.apply_async(entry_args, entry_kwargs, producer=producer, **entry.options)
```

Comment:
Is there anything we need to do with the `entry.options` in the Kubernetes case?

Reply:
This is one of the example `entry.options` it seems like we encapsulated all the information in the job_result we created above.
```
{'nautobot_job_scheduled_job_id': UUID('320bc2c1-6e98-48db-abd7-7c58c26fdd26'), 'headers': {}, 'nautobot_job_user_id': UUID('970ef2cc-d8e8-456f-90f2-1c45bac440ec'), 'nautobot_job_job_model_id': UUID('80f97ef5-5dfe-4bd7-8c2a-dafdb0b12870')}
```